saleor/graphql/app/mutations.py

```diff
@@ -208,6 +211,75 @@ def clean_instance(cls, info, instance):
             raise ValidationError({"id": ValidationError(msg, code=code)})
 
 
+class AppActivate(ModelMutation):
+    class Arguments:
+        id = graphene.ID(description="ID of app to activate.", required=True)
+
+    class Meta:
+        description = "Activate the app."
+        model = models.App
+        permissions = (AppPermission.MANAGE_APPS,)
+        error_type_class = AppError
+        error_type_field = "app_errors"
+
+    @classmethod
+    def clean_instance(cls, info, app: App):
+        requestor = get_user_or_app_from_context(info.context)
+        permissions = app.permissions.all()
+        if not requestor_is_superuser(requestor) and not requestor.has_perms(
+            permissions
+        ):
+            msg = "You don't have enough permission to perform this action."
+            code = AppErrorCode.OUT_OF_SCOPE_APP.value
+            raise ValidationError({"id": ValidationError(msg, code=code)})
+
+        if app.is_active:
+            msg = "App is already active."
+            code = AppErrorCode.INVALID.value
+            raise ValidationError({"id": ValidationError(msg, code=code)})
```

Comment:
I'm wondering, is it useful to raise this validation error? What if the mutation just did nothing and returned the unmodified object?